### ISSUE TITLE
Data modeling

### ISSUE THREAD

**Issue by denisemauldin** [devroastproject/devroastproject#18]
Determine the data models for the project based on the README information about the site purpose

Poll or Feedback or Post
Comment

Are Votes a separate resource or are they attributes of a Poll or a Comment?

How do Poll/Post/Feedback relate to Comment and Vote?

https://docs.djangoproject.com/en/3.1/topics/db/models/

Use the docs to figure out what kind of fields each piece of data should be stored as in the database.

**Comment by D-Bits** [comment]
I would be interested in working on this.

**Comment by asgray** [comment]
1st attempt at an ERD for the database. I focused mostly on entity types and left out details like datatypes for now.
[Devroast_ERD_v1.pdf](https://github.com/devroastproject/devroastproject/files/5177511/Devroast_ERD_v1.pdf)

**Comment by crg253** [comment]
Thanks for doing this. I'll look at it tomorrow.

**Comment by denisemauldin** [comment]
Ok looks like this is using `Project`, `Question` and `Comment` as the main three models. 

What's the difference between a `Question` and a `Comment`?

Django has already created the `auth_user` table, which typically uses a foreign key of `user_id` in the other tables.  Django is particular about the naming of app/table so they might be slightly different than what's in the ERD, but generally I think it looks good.

**Comment by denisemauldin** [comment]
This ERD is modeled like this:

class Project:
  user = ...
  title = ...
  repo_url = ...
  hosted_url = ...

class Comment:
  post = Project.id
  user = ...
  body = ...
  neg_votes = ...
  pos_votes = ...
  parent = models.ForeignKey('self', related_name='reply_set', null=True)


So if the goal is to get all of the comments, you would do:
Project.objects.get(id=5).comment_set.all() to get the first level of comments
then to get the levels under the first comment, you'd have to do a for loop through all the comment_set I think?
for comment in Project.objects.get(id=5).comment_set.all():
  replies = comment.reply_set.all()

Can anyone think of a better way to get the 2nd level of comments?

**Comment by denisemauldin** [comment]
If the ERD instead had one Project that had one Comment:

class Project:
  user = ...
  title = ...
  repo_url = ...
  hosted_url = ...
  comment = Comment.id

class Comment:
  user = ...
  body = ...
  neg_votes = ...
  pos_votes = ...
  parent = models.ForeignKey('self', related_name='reply_set', null=True)

In this you would get the first Comment with:
Project.objects.get(id=5).comment
and you'd get the rest of the comments with:
Project.objects.get(id=5).comment.reply_set.all()

**Comment by D-Bits** [comment]
I'm think Project should have many comments, and that Comment should be its own model.

**Comment by asgray** [comment]
I agree with 1 Project : N Comments. Comments have enough separate attributes and behavior to warrant it.

**Comment by denisemauldin** [comment]
Team decision is to go with a Project has many Comments

**Comment by denisemauldin** [comment]
When discussing Tags, the team decide to have a link both on Project to Tag and Comment to Tag so that the OP can post a project for feedback with tags and without a comment, and also so that an OP can post a project for feedback with specific questions (Comment) that have tags.

**Comment by denisemauldin** [comment]
@D-Bits Please add a link between Project and Tag in the PR #31
